demo/summer-2016.html

```diff
@@ -253,30 +253,38 @@
     }
     editor.commands.addCommand(keyboardHandler);
 
-    function isChangePosition() {
+
+    function removeHighlightBracket() {
       if (session.$bracketHighlightRight || session.$bracketHighlightLeft) {
         session.removeMarker(session.$bracketHighlightLeft);
         session.removeMarker(session.$bracketHighlightRight);
+        session.$highlightMode = false;
         session.$bracketHighlightLeft = null;
         session.$bracketHighlightRight = null;
       }
     }
 
-    document.getElementById("editor").addEventListener("click", function(){
-      //TODO: need a crossed check whether a highlight border
-
-    });
-    document.getElementById("editor").addEventListener("keyup", function(event){
-      if (event.keyCode <= 40 && event.keyCode >= 37) {
-        //TODO: need a crossed check whether a highlight border
-      } else if (!(event.keyCode == 57 && event.ctrlKey)) {
-        console.log("in else");
-        isChangePosition();
-        highlighter.highlightBrackets(editor);
-      }
+    session.getSelection().on("changeCursor", function() {
+      if (session.$highlightMode)
+        if (isOutFromAreaOfHighlight())
+          removeHighlightBracket();
+        else {
+          removeHighlightBracket();
+          highlighter.highlightBrackets(editor);
+        }
     });
 
-
+    function isOutFromAreaOfHighlight() {
+      var oldPosition = {};
+      for(var key in session.$positionOfHighlight)
+        oldPosition[key] = session.$positionOfHighlight[key];
```

Comment:
btw, the only value which you assign to $positionOfHoghlight is your `result` structure with left/right brackets and mismatch. Are you iterating these fields? What's the point of this loop?